app/controllers/admin/admin_controller.rb

```diff
@@ -1,16 +1,26 @@
 class Admin::AdminController < ApplicationController
+  # include UsersHelper # Perhaps need to create the same method in AdminHelper and include it?
+
   def index
     @users = User.find(:all, :conditions => ["admin = ?", false])
     authorize current_user
   end
 
   def show
-    @user = User.find(params[:id])
+    @user = User.find_by_id(params[:id])
+    if @user.nil?
```

Comment:
A better way to do check and reroute. http://stackoverflow.com/questions/2385799/how-to-redirect-to-a-404-in-rails